### PR TITLE
Autodoc Improvement

### PR DESCRIPTION
**Pull request by Surrealaser**
-Can now see the occupant of the auto-doc, including most recent medical records with a medHUD via examine.

-Can now scan the occupant of an auto-doc by using a health analyzer on it.

-Shrapnel Removal Surgery renamed to Foreign Body Removal Surgery; it now scans for and removes xeno embryos in addition to shrapnel. This is a necessary remedy due to the limitations of low-pop, beyond the fact that we don't go by first contact strictures.

-Medic skill level marines are no longer subject to delays for using/inserting people into the auto-doc, allowing them to more efficiently assist/fill in during low pop.

### CODE REVIEW

File: code/game/machinery/autodoc.dm [hunk 1]
```diff
@@ -433,7 +439,7 @@
 						close_incision(H,S.limb_ref)
 
 					if("shrapnel")
-						if(prob(30)) visible_message("\The [src] speaks, Beginning shrapnel removal.");
+						if(prob(30)) visible_message("\The [src] speaks, Beginning foreign body removal.");
```
[the remark this answers is not in the document]
Comment (Surrealaser) [reply code/game/machinery/autodoc.dm]: Was the existing convention

File: code/game/machinery/adv_med.dm [hunk 2]
```diff
@@ -228,25 +232,14 @@
 	var/dat
 	if (delete && temphtml) //Window in buffer but its just simple message, so nothing
 		delete = delete
+		to_chat(world, "\icon[src]<span class='warning'>SCANNER DEBUG: DELETE TRIGGERED</span>")
```
Comment (Rohesie) [inline code/game/machinery/adv_med.dm]: We need these gone in order to test it.

File: code/game/machinery/adv_med.dm [hunk 3]
```diff
@@ -274,205 +265,19 @@
 		R.name = "Body scan report -[src.connected.occupant.real_name]-"
 		R.info = format_occupant_data(src.connected.get_occupant_data())
 
-
-/obj/machinery/bodyscanner/proc/get_occupant_data()
-	if (!occupant || !istype(occupant, /mob/living/carbon/human))
+/obj/machinery/bodyscanner/examine(mob/living/user)
+	..()
```
Comment (Rohesie) [inline code/game/machinery/adv_med.dm]: `. = ..()`

File: code/game/machinery/autodoc.dm [hunk 4]
```diff
@@ -167,10 +167,16 @@
...
 				surgery_list += create_autodoc_surgery(L,LIMB_SURGERY,"necro")
+			var/skip_embryo_check = FALSE
 			if(L.implants.len)
 				for(var/I in L.implants)
 					if(!is_type_in_list(I,known_implants))
 						surgery_list += create_autodoc_surgery(L,LIMB_SURGERY,"shrapnel")
+						if(L.name == "chest")
+							skip_embryo_check = TRUE
+			var/obj/item/alien_embryo/A = locate() in M
+			if(A && L.name == "chest" && !skip_embryo_check) //If we're not already doing a shrapnel removal surgery proceed.
```
[the remark this answers is not in the document]
Comment (Rohesie) [reply code/game/machinery/autodoc.dm]: ?

File: code/game/machinery/autodoc.dm [hunk 5]
```diff
@@ -630,16 +655,45 @@
...
-/obj/machinery/autodoc/proc/go_out()
-	if(!occupant) return
+/obj/machinery/autodoc/proc/go_out(notice_code = FALSE)
+	if(!occupant)
+		return
 	occupant.forceMove(loc)
+	if(connected.release_notice) //If auto-release notices are on as they should be, let the doctors know what's up
+		var/reason = "Reason for discharge: Procedural completion."
+		switch(notice_code)
+			if(1)
```
Comment (Rohesie) [inline code/game/machinery/autodoc.dm]: Magic numbers, use defines.

Comment (Surrealaser) [reply code/game/machinery/autodoc.dm]: ?

You want me to use defines for the notice code?

Really? A separate define for each event type?

Comment (Rohesie) [reply code/game/machinery/autodoc.dm]: Yes, that's the idea. Unless @Qustinnus disagrees here?

Comment (Qustinnus) [reply code/game/machinery/autodoc.dm]: @Surrealaser yes because i cant be arsed finding out what those magic numbers mean

Comment (Qustinnus) [reply code/game/machinery/autodoc.dm]: its essentialy an enum now but without the identifier which makes me fucking weep every time i review someones code and I have to extrapolate what the fuck 1 to 17 means.

**Pull request merged**